System/Directory/Internal/Windows.hsc

```diff
@@ -579,10 +579,13 @@ setTimes :: FilePath -> (Maybe POSIXTime, Maybe POSIXTime) -> IO ()
 setTimes path' (atime', mtime') =
   bracket (openFileHandle path' Win32.gENERIC_WRITE)
           Win32.closeHandle $ \ handle ->
+#if MIN_VERSION_Win32(2,12,0)
+  Win32.setFileTime handle Nothing (posixToWindowsTime <$> atime') (posixToWindowsTime <$> mtime')
+#else
   maybeWith with (posixToWindowsTime <$> atime') $ \ atime'' ->
   maybeWith with (posixToWindowsTime <$> mtime') $ \ mtime'' ->
-  Win32.failIf_ not "" $
```

Comment:
What happened to the `failIf_`?

Reply:
Thank you for catching this! This would have been terrible had it snuck in.